### ISSUE TITLE
16.60 doesn't build from AUR w/ pacaur

### ISSUE THREAD

**Issue by corngood** [corngood/archlinux-amdgpu#30]
A couple of people have reported an error when running `pacaur -S amdgpu-pro`:

```
:: resolving dependencies...
:: dependency cycle detected (amdgpu-pro=16.60.379184-1 amdgpu-pro-dkms=16.60.379184-1 amdgpu-pro-libdrm=16.60.379184-1 amdgpu-pro-opencl=16.60.379184-1 amdgpu-pro-vdpau=16.60.379184-1 amdgpu-pro-vulkan=16.60.379184-1 lib32-amdgpu-pro=16.60.379184-1 lib32-amdgpu-pro-libdrm=16.60.379184-1 lib32-amdgpu-pro-opencl=16.60.379184-1 lib32-amdgpu-pro-vdpau=16.60.379184-1 lib32-amdgpu-pro-vulkan=16.60.379184-1 xf86-video-amdgpu-pro=16.60.379184-1)
```

Looks like the cycle is because of things depending on `amdgpu-pro`?  @brainpower do you see anything wrong with the dependencies?

**Comment by nyorain** [comment]
amdgpu-pro depends on xf86-video-amdgpu-pro and vice versa.
There are more cycle dependencies, am already working on it, shouldn't be that hard to fix.

**Comment by JontySR** [comment]
That issue can easily be bypassed with 'git clone' and 'makepkg -i', as long as xorg-server and xorg-server-common are downgraded to 1.18.0-1, and xf86-input-libinput is downgraded to 0.15.0-2 of course. When installed though, there is corrupted glitchy text.

I'm on an HD 7850, using GNOME 3.22 and linux 4.9.7.

![screenshot from 2017-02-07 20-52-48](https://cloud.githubusercontent.com/assets/25610245/22711004/7607d474-ed77-11e6-8493-9b60d7f4e7dd.png)

**Comment by nyorain** [comment]
nice! got it working now too, don't have any text problems.
Do you have sound? can't get that to work (can only test with hdmi but that should work with amdgpu pro shouldn't it?)

**Comment by JontySR** [comment]
I have sound, but don't pass it through HDMI. I suppose it might be an issue carried over from remnants of the open-source driver. Glad you don't have the graphical glitches though. Did you set up any config stuff?

**Comment by nyorain** [comment]
nope didn't set anything up, also tested with nautilus. And oh that are bad news. Hoped it would work with the pro driver...
![test2](https://cloud.githubusercontent.com/assets/13062371/22713760/065d94d4-ed8a-11e6-8274-f877b92beb20.png)

**Comment by JontySR** [comment]
I'll probs switch to ubuntu budgie or something while I have this card in my computer, but I'll happily test any more stuff that gets put out.

**Comment by cra0zy** [comment]
You guys got it to build? For me dksm failed to build it :|
```
==> dkms install amdgpu-pro-16.60/379184 -k 4.9.8-1-ARCH
Error! Bad return status for module build on kernel: 4.9.8-1-ARCH (x86_64)
Consult /var/lib/dkms/amdgpu-pro-16.60/379184/build/make.log for more information.
```

Log: https://gist.github.com/cra0zy/5fc997de93cf8847e9584f2a66ef3923

**Comment by fjfalcon** [comment]
Just to notice: i have working hdmi sound without any problems with amdgpu pro.

**Comment by JontySR** [comment]
@cra0zy It looks like it doesn't like your version of the kernel, which is only available in testing. Can you confirm that this is the issue @corngood ?

> Just to notice: i have working hdmi sound without any problems with amdgpu pro.
From @fjfalcon so I suppose there is hope, @nyorain 
You should probably open a separate issue for HDMI audio then.

**Comment by nyorain** [comment]
yeah thanks for the notice there is hope again now 👍 
Probably something wrong with my audio config then. Will open an issue if i can't get it to work with the driver.

**Comment by corngood** [comment]
Anyone else have a GCN 1.0 card like @JontySR, and if so, do you have similar corruption?

@cra0zy I'm not sure why only you are getting that error.  It doesn't look like it's related to kernel changes.  Do you have an unusual gcc?

**Comment by cra0zy** [comment]
No unusual gcc, but I do have testing repo enabled (technically had since I deleted my arch linux, totally unrelated to this).

**Comment by stonewalljones** [comment]
I still get dependency issues if I try makepkg -i
`stonewall@axiom ~/build/amdgpu-pro-installer (git)-[master] % makepkg -i                                                                                                                                                                 :(
==> WARNING: The package group has already been built, installing existing packages...
==> Installing amdgpu-pro-installer package group with pacman -U...
loading packages...
resolving dependencies...
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "amdgpu-pro-vdpau=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "amdgpu-pro-vdpau=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro", a dependency of "amdgpu-pro-libgl"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "amdgpu-pro-vdpau=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "xf86-video-amdgpu-pro"
warning: cannot resolve "xf86-video-amdgpu-pro=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "amdgpu-pro-vdpau=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "xf86-video-amdgpu-pro"
warning: cannot resolve "xf86-video-amdgpu-pro=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro", a dependency of "lib32-amdgpu-pro-libgl"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro-vdpau"
warning: cannot resolve "lib32-amdgpu-pro-vdpau=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro-vdpau"
warning: cannot resolve "amdgpu-pro-vdpau=16.60.379184-1", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-ncurses5-compat-libs>=6", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "ncurses5-compat-libs>=6", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "xf86-video-amdgpu-pro"
:: The following packages cannot be upgraded due to unresolvable dependencies:
      amdgpu-pro  amdgpu-pro-libgl  amdgpu-pro-vdpau  lib32-amdgpu-pro  lib32-amdgpu-pro-libgl  lib32-amdgpu-pro-vdpau  xf86-video-amdgpu-pro

:: Do you want to skip the above packages for this upgrade? [y/N] n
error: failed to prepare transaction (could not satisfy dependencies)
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro-vdpau=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro-vdpau=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: amdgpu-pro-libgl: requires amdgpu-pro
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro-vdpau=16.60.379184-1
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-ncurses5-compat-libs>=6
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: xf86-video-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro: requires xf86-video-amdgpu-pro=16.60.379184-1
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro-vdpau=16.60.379184-1
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-ncurses5-compat-libs>=6
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: xf86-video-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro: requires xf86-video-amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro-libgl: requires lib32-amdgpu-pro
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: lib32-amdgpu-pro-vdpau: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro-vdpau=16.60.379184-1
:: amdgpu-pro-vdpau: requires ncurses5-compat-libs>=6
:: lib32-amdgpu-pro: requires amdgpu-pro-vdpau=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-ncurses5-compat-libs>=6
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: amdgpu-pro: requires ncurses5-compat-libs>=6
:: xf86-video-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
==> WARNING: Failed to install built package(s).
`

**Comment by corngood** [comment]
@stonewalljones I think you're just missing `lib32-ncurses5-compat-libs` and `ncurses5-compat-libs`?

**Comment by stonewalljones** [comment]
Ok I get a much smaller list now
```

stonewall@axiom ~/build/amdgpu-pro-installer (git)-[master] % makepkg -i                                                     
==> WARNING: The package group has already been built, installing existing packages...
==> Installing amdgpu-pro-installer package group with pacman -U...
loading packages...
resolving dependencies...
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro", a dependency of "amdgpu-pro-libgl"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro", a dependency of "lib32-amdgpu-pro-libgl"
warning: cannot resolve "lib32-libomxil-bellagio", a dependency of "lib32-amdgpu-pro"
warning: cannot resolve "lib32-amdgpu-pro=16.60.379184-1", a dependency of "amdgpu-pro"
warning: cannot resolve "amdgpu-pro=16.60.379184-1", a dependency of "xf86-video-amdgpu-pro"
:: The following packages cannot be upgraded due to unresolvable dependencies:
      amdgpu-pro  amdgpu-pro-libgl  lib32-amdgpu-pro  lib32-amdgpu-pro-libgl  xf86-video-amdgpu-pro

:: Do you want to skip the above packages for this upgrade? [y/N] n
error: failed to prepare transaction (could not satisfy dependencies)
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: amdgpu-pro-libgl: requires amdgpu-pro
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: lib32-amdgpu-pro-libgl: requires lib32-amdgpu-pro
:: lib32-amdgpu-pro: requires lib32-libomxil-bellagio
:: amdgpu-pro: requires lib32-amdgpu-pro=16.60.379184-1
:: xf86-video-amdgpu-pro: requires amdgpu-pro=16.60.379184-1
==> WARNING: Failed to install built package(s).

```

Also what happens if I do not downgrade X11?
I just saw in the discussion above that you are supposed  to do that.

**Comment by corngood** [comment]
@stonewalljones Yeah, you'll need to install `lib32-libomxil-bellagio` from AUR as well.

You really need to downgrade Xorg to 1.18.  The binary driver will not even start with 1.19 unless you disable ABI version checking, and then it will probably crash.

@JontySR listed the relevant packages in https://github.com/corngood/archlinux-amdgpu/issues/31 if that helps.  I'll try to get a concise explanation of all of this on the AUR page once the dependencies are fixed.

**Comment by nyorain** [comment]
@fjfalcon Did you configure anything for amdgpu? reinstalled arch and can't get hdmi audio with pulse or alsa directly to work...

I think one of the problems in building the pacakge from aur is currently that some of its dependencies are broken pacakges. I could not install the ncurses and the libomxil package using pacaur and had to use makepkg manually (with --skippgpcheck). 
And for the depedency cycle couldn't we simply remove the core packages (i.e. [lib32-]amdgpu-pro) from the dependencies of all other packages?
Would also be really useful if the PKGBUILD downgraded the xorg packages automatically...

**Comment by JontySR** [comment]
@nyorain nurses wasn't broken, the package info page said to look in the PKGBUILD for a key and add it, which worked. I may have had issues with some of the other packages, but I can't remember.

**Comment by nyorain** [comment]
oh right, never stumbled across this when installing a package from aur. Would be useful to mention in installation instructions nontheless i guess.

**Comment by corngood** [comment]
@nyorain is it possible to downgrade xorg automatically?  The best solution I can think of is to give the same advice as:

**Comment by HanFox** [comment]
I would be very wary of forcing any sort of downgrade. As people continually argue with the AUR packages people should know what they're doing when they install things. Packages shouldn't "take over" in any way.

The best place to put the information for install is on the archwiki. That's what it's for.

**Comment by JontySR** [comment]
@corngood I think I got rid of the text corruption. I ended up reinstalling Arch and this time building up purely from xorg 1.18 and the driver. Currently on lxqt and haven't noticed any glitches. Will keep you posted if it changes though.

**Comment by fjfalcon** [comment]
@nyorain nope, everything from default. Default pulseaudio setup.

**Comment by brainpower** [comment]
This should get rid of dependency cycles: https://github.com/corngood/archlinux-amdgpu/pull/33
Please review or test.

**Comment by nyorain** [comment]
@brainpower tested with your branch, makepkg -i still gives me
```
warning: dependency cycle detected:
warning: amdgpu-pro-libgl will be installed before its amdgpu-pro dependency
warning: dependency cycle detected:
warning: lib32-amdgpu-pro-libgl will be installed before its lib32-amdgpu-pro dependency
```
That are at least way less warnings than before. Thanks for your work btw.

@fjfalcon That is very strange. I can only get dummy output with pulse. Pavucontrol lists all hdmi connectors as unplugged and unavailable. Maybe something wrong with my config but i reinstalled arch today and it works on radeon without any amdgpu stuff... Got no further ideas. Do you use the driver package from this repo with all patches appplied?

**Comment by corngood** [comment]
I just pushed a new pkgrel with #33.  `pacaur -S amdgpu-pro` is no longer going to install all the packages.  You'll need to specify the packages you want (libgl, lib32-, etc).

I'm guessing it still has the warnings @nyorain mentioned, so I'll leave this open for now.

**Comment by mikel392** [comment]
this is still an issue for me 

warning: dependency cycle detected:
warning: amdgpu-pro-libgl will be installed before its amdgpu-pro dependency
warning: dependency cycle detected:
warning: lib32-amdgpu-pro-libgl will be installed before its lib32-amdgpu-pro dependency
warning: dependency cycle detected:
warning: xserver-xorg-video-amdgpu-pro will be removed after its amdgpu-pro-graphics dependency

Packages (20) amdgpu-pro-graphics-16.30.3.315407-2 [removal]  lib32-amdgpu-pro-lib32-16.30.3.315407-2 [removal]  lib32-cdparanoia-10.2-2  lib32-gst-plugins-base-1.8.3-1  lib32-libvisual-0.4.0-5
              xserver-xorg-video-amdgpu-pro-16.30.3.315407-2 [removal]  amdgpu-pro-16.60.379184-2  amdgpu-pro-dkms-16.60.379184-2  amdgpu-pro-libdrm-16.60.379184-2  amdgpu-pro-libgl-16.60.379184-2
              amdgpu-pro-opencl-16.60.379184-2  amdgpu-pro-vdpau-16.60.379184-2  amdgpu-pro-vulkan-16.60.379184-2  lib32-amdgpu-pro-16.60.379184-2  lib32-amdgpu-pro-libdrm-16.60.379184-2
              lib32-amdgpu-pro-libgl-16.60.379184-2  lib32-amdgpu-pro-opencl-16.60.379184-2  lib32-amdgpu-pro-vdpau-16.60.379184-2  lib32-amdgpu-pro-vulkan-16.60.379184-2  xf86-video-amdgpu-pro-16.60.379184-2

Total Installed Size:  498.92 MiB
Net Upgrade Size:      464.11 MiB

:: Proceed with installation? [Y/n] 
(17/17) checking keys in keyring                                                                                                [#############################################################################] 100%
(12/17) checking package integrity                                                                                              [#############################################################################] 100%
(12/17) loading package files                                                                                                   [#############################################################################] 100%
(17/17) checking for file conflicts                                                                                             [#############################################################################] 100%
error: failed to commit transaction (conflicting files)
amdgpu-pro: /etc/amd/amdapfxx.blb exists in filesystem
amdgpu-pro: /usr/lib/dri/amdgpu_dri.so exists in filesystem
amdgpu-pro: /usr/share/doc/libegl1-amdgpu-pro/changelog.Debian.gz exists in filesystem
amdgpu-pro: /usr/share/doc/libegl1-amdgpu-pro/copyright exists in filesystem
amdgpu-pro: /usr/share/doc/libgbm1-amdgpu-pro/changelog.Debian.gz exists in filesystem
amdgpu-pro: /usr/share/doc/libgbm1-amdgpu-pro/copyright exists in filesystem
amdgpu-pro: /usr/share/doc/libgl1-amdgpu-pro-dri/changelog.Debian.gz exists in filesystem
amdgpu-pro: /usr/share/doc/libgl1-amdgpu-pro-dri/copyright exists in filesystem
amdgpu-pro: /usr/share/doc/libgl1-amdgpu-pro-glx/changelog.Debian.gz exists in filesystem
amdgpu-pro: /usr/share/doc/libgl1-amdgpu-pro-glx/copyright exists in filesystem
amdgpu-pro: /usr/share/doc/libgles2-amdgpu-pro/changelog.Debian.gz exists in filesystem
amdgpu-pro: /usr/share/doc/libgles2-amdgpu-pro/copyright exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm-amdgpu-pro-amdgpu1/changelog.Debian.gz exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm-amdgpu-pro-amdgpu1/copyright exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm-amdgpu-pro-dev/changelog.Debian.gz exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm-amdgpu-pro-dev/copyright exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm2-amdgpu-pro/changelog.Debian.gz exists in filesystem
amdgpu-pro-libdrm: /usr/share/doc/libdrm2-amdgpu-pro/copyright exists in filesystem
amdgpu-pro-opencl: /etc/OpenCL/vendors/amdocl64.icd exists in filesystem
amdgpu-pro-vdpau: /usr/share/doc/libvdpau-amdgpu-pro/changelog.Debian.gz exists in filesystem
amdgpu-pro-vdpau: /usr/share/doc/libvdpau-amdgpu-pro/copyright exists in filesystem
lib32-amdgpu-pro: /usr/lib32/dri/amdgpu_dri.so exists in filesystem
lib32-amdgpu-pro-opencl: /etc/OpenCL/vendors/amdocl32.icd exists in filesystem
Errors occurred, no packages were upgraded.

**Comment by brainpower** [comment]
To fix the "exists in filesystem" errors, remove any previous version of amdgpu-pro packages.
Since the structure of the packages changed with the new version, the new version would have to conflict and provide or replace every single one of the many split packages the previous version had.
To figure all those conflicts, provides and replaces correctly is a lot harder than removing the old packages before upgrading.

**Comment by mikel392** [comment]
@brainpower ah alright thank you, will do that.

**Comment by Yogosh** [comment]
@JontySR Had the same bug on kernel 4.9.8-1, downgrading to 4.8.9-1 fixed the problem.

**Comment by corngood** [comment]
Closing because 17.10 is out.  17.30 hopefully coming soon.
